### DISCUSSION TITLE
Unsure how to implement drag boundary so pieces can't be dragged offstage.

### DISCUSSION THREAD

**Discussion by jwotman** [flbulgarelli/headbreaker#40] [Q&A] [answered]
I find your framework really interesting and helpful.  Plan to use it for a kids section of a site I am working on.   Noticed that pieces can be dragged offstage and if a group of connected pieces are dragged partially offstage the result is stacked pieces at the edge.   I would like to make it impossible to drag anything offstage.  Do you have any suggestions?

**Comment by flbulgarelli** [comment, marked as answer]
Hi!

Glad to hear that :smile: 

I am currently facing the same issue so I recently made the canvas draggable by default - you can however disable this by using the `fixed` property of [Canvas](https://flbulgarelli.github.io/headbreaker/api/Canvas.html). I have just updated the demo to reflect that behavior. I will deploying it before next week.  

I think it could be nicer to implement a bouncing behavior, but I don't believe I will be working on that in the next few weeks, so the draggable stage is the best solution I can offer right now. 

To get a better visual result, I'd suggest you adding a semi-traslucid layer on the borders of the canvas, so that your users can easily notice that their pieces are going offstage. I may add that feature to the framework in the future, but on the meanwhile I am probably leaving out of scope.

**Comment by jwotman** [comment]
That's workable for me in the short-term.  I'll put an image border on a container div and see how that goes with the kids.  I like the idea of the bouncing behavior.   Once again, thanks for this framework.  I've looked at a lot of GitHub Jigsaw apps and yours was definitely the most useful and extensible.

**Comment by nabiloo19** [comment]
It is not working on my end. They still float out the border. Please help?!
